draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+#### Exclusion of "$comment"
+
+For compatibility with JSON Schema {{?I-D.bhutton-json-schema-01, Section 8.3}}, JRI requires that a `"$comment"` keyword present in the same object as any JRI keywords MUST NOT impact the semantics of those JRI keywords.  In particular, `"$comment"` MUST NOT be used as a semantic disambiguator as described in {{disambiguator}}.
+
+However, `"$comment"` is not itself considered to be a JRI keyword.  JRI does not place any requirements on the usage, data type, syntax, or other semantics of `"$comment"` in context specifications.  JRI implementations MUST NOT require that a `"$comment"` adjacent to JRI keywords conform to JSON Schema's `"$comment"` keyword, unless explicitly required to do so by a context specification.
+
+### Evaluation order
+
+JRI behaviors, whether implemented by JRI keywords or by context specification features, MUST be evaluated in the following order when present:
+
+1. Primary resource identifiers (`"$id"`)
+1. Secondary resource identifiers (`"$anchor"`)
+1. Locations (`"$defs"`)
+1. References (`"$ref"`, `"$extRef"`)
+
+The logic of this ordering is that primary resources must be identified first as all other keywords depend having the correct base IRI.  Next, secondary resources can be identified once the primary resource is known.  Finally, nested locations which can be reference targets or have additional identifier keywords can be discovered within the primary resource.
+
+### Identification
+
+JRI offers two identification keywords:  one for primary resources, and one for secondary resources.
+
+#### `"$id"`
+
+The value of the `"$id"` property MUST be a string, and MUST be a valid IRI-reference as defined by {{!RFC3987, Section 2.2}}, and MUST NOT contain a fragment.
+
+The object containing the `"$id"` property MUST be considered to be a primary resource and to be identified by the IRI produced by resolving the IRI-reference against the current base IRI.  This IRI MUST be considered the base IRI for the newly identified primary resource, in accordance with {{!RFC3986, Section 5.1.1}}.
+
+If the object containing the `"$id"` property is not the root object of the document, the encapsulating resource MUST be considered to be an "encapsulating entity" per {{!RFC3986, Section 5.1.2}}.
+
+#### `"$anchor"`
+
+The value of the `"$anchor"` property MUST be a string, and MUST be a valid IRI fragment according to the `ifragment` ABNF production in {{!RFC3987, Section 2.2}}.
+
+### Location
+
+Location behavior is necessary when not all locations within a data format are valid reference targets, or when object properties that appear to be JRI keywords are only actually JRI keywords in certain locations.  In formats where most locations cause automatic actions (such as applying a schema to an instance), location behavior can also create areas where the automatic actions do not apply.  These reserved locations can safely hold re-usable objects separate from any single use.
```

Comment:
@DavidBiesack thanks, these sections used to be in a different order and I missed this, will fix and ping you once I have!  I appreciate the careful reading.

Reply:
I still think this section is not clear enough to someone coming to this draft without background.
Define "Location behavior" above (new term?), or perhaps use "location-specific behavior" which I think is clearer.

In some data formats, not all possible locations are valid reference targets, or object properties that appear to be JRI keywords are only actually JRI keywords in certain locations. In such cases, _location-specific behavior_ is necessary. In formats where most locations cause automatic actions (such as applying a schema to an instance), location behavior can also define areas where the automatic actions do not apply. ...

(I suggest using "define" instead of "create" - the format does not "create" locations in a document.